Plugins/Flow.Launcher.Plugin.Explorer/Search/DirectoryInfo/DirectoryInfoSearch.cs

```diff
@@ -58,38 +64,38 @@ private List<Result> DirectorySearch(SearchOption searchOption, Query query, str
             {
                 var directoryInfo = new System.IO.DirectoryInfo(path);
 
-                foreach (var fileSystemInfo in directoryInfo.EnumerateFileSystemInfos(searchCriteria, searchOption))
+                foreach (var fileSystemInfo in directoryInfo.EnumerateFileSystemInfos(searchCriteria, enumerationOption))
                 {
-                    if ((fileSystemInfo.Attributes & FileAttributes.Hidden) == FileAttributes.Hidden) continue;
-
                     if (fileSystemInfo is System.IO.DirectoryInfo)
                     {
-                        folderList.Add(resultManager.CreateFolderResult(fileSystemInfo.Name, fileSystemInfo.FullName, fileSystemInfo.FullName, query, true, false));
+                        folderList.Add(resultManager.CreateFolderResult(fileSystemInfo.Name, fileSystemInfo.FullName,
+                            fileSystemInfo.FullName, query, true, false));
                     }
                     else
                     {
                         fileList.Add(resultManager.CreateFileResult(fileSystemInfo.FullName, query, true, false));
                     }
+
+                    if (token.IsCancellationRequested)
+                        return null;
```

Comment:
in your previous changes i have seen you use ThrowIfCancellationRequested() and catch it with an OperationCancelledException. Is one way better than the other? Should we use IsCancellationRequested way to handle it more elegantly?

Reply:
Agree with you. I will change that to throw, which can reduce some code for checking null.

Reply:
Well I thought iscancellationrequested is nicer and instead of returning null you can just return empty results? Rather than having an extra catch.

Is there a reason for returning null rather than empty result list?

Reply:
Just a few difference, not a lot.
We will just throw the error, which will be catched by PluginManager.QueryForPluginAsync, so no worry for doing null check.

Reply:
Yeah but cancellation of previous query is a mechanism we require throughout, so probably not great to be treating it like the program errored out. Maybe change this next time in separate pr, for now just leave it as cancellationrequested

Reply:
Well actually, throw an taskcancelledexcpetion is what Microsoft suggested in cooperative cancellation. This is similar to how we shall handle operationcancelledexception, which is the base of taskcancelledexcpetion.

However, it seems that in Debug mode, throwing a lot of exception may cause stunt……which is not a bit due, but somehow not promising.